### PR TITLE
allow incident update to update component status, bug fix #2767

### LINKED ISSUE #2767: Updating incident status does not update component
On 2.4

1. create an incident, setting its status to "investigating" and the component to e.g. partial outage
 --> the component selected moves to "partial outage" status
2. update the incident, setting it as fixed
3. there is no possibility to set as "functional the component" 
 --> the component remains as "partial outage"

Is it planned to give this possibility too? It would make sense to me.
Thanks,
Michele

### PR DESCRIPTION
**Pull request by imknight**

### CODE REVIEW

File: app/Http/Controllers/Dashboard/IncidentController.php [hunk 1]
```diff
@@ -321,6 +321,10 @@ public function createIncidentUpdateAction(Incident $incident)
                 ->withErrors($e->getMessageBag());
         }
 
+        if ($incident->incident->component) {
+            $incident->incident->component->update(['status' => Binput::get('component_status')]);
```
Comment (jbrooksuk) [inline app/Http/Controllers/Dashboard/IncidentController.php]: The component and the status should be passed through to the `CreateIncidentUpdateCommand` :)

Comment (imknight) [reply app/Http/Controllers/Dashboard/IncidentController.php]: noted, updated my commit.

**Comment by xBytez** [comment]
+1, I like this change.

File: app/Http/Controllers/Dashboard/IncidentController.php (hunk 1 above)
Comment (jbrooksuk) [reply app/Http/Controllers/Dashboard/IncidentController.php]: These lines need moving into the `CreateIncidentUpdateCommandHandler`.

**Comment by jbrooksuk** [comment]
Can you drop the commit about StyleCI please, that will happen automatically later on :)

**Comment by jbrooksuk** [comment]
Can you rebase? We'll get this merged.

**Comment by jbrooksuk** [comment]
Replaced by https://github.com/CachetHQ/Cachet/pull/2884

**Pull request closed without merging**